pandas/core/indexes/interval.py

```diff
@@ -1102,11 +1102,8 @@ def func(self, other, sort=True):
                        'objects that have compatible dtypes')
                 raise TypeError(msg.format(op=op_name))
 
-            if op_name == 'difference':
-                result = getattr(self._multiindex, op_name)(other._multiindex,
-                                                            sort)
-            else:
-                result = getattr(self._multiindex, op_name)(other._multiindex)
+            result = getattr(self._multiindex, op_name)(other._multiindex,
```

Comment:
Now all the set operations should be implemented for ``MultiIndex`` so I can get rid of the ``if`` statement